gobblin-salesforce/src/main/java/org/apache/gobblin/salesforce/SalesforceSource.java

```diff
@@ -131,7 +136,19 @@
     Partition partition = new Partitioner(state).getGlobalPartition(previousWatermark);
     Histogram histogram = getHistogram(sourceEntity.getSourceEntityName(), watermarkColumn, state, partition);
 
-    String specifiedPartitions = generateSpecifiedPartitions(histogram, minTargetPartitionSize, maxPartitions,
+    // we should look if the count is too big, cut off early if count exceeds the limit, or bucket size is too large
+    Histogram histogramActual = new Histogram();
```

Comment:
We should consider pushing this logic into getHistogram() for early termination of the search or add a note here indicating possible future work.